x-pack/plugins/lens/public/xy_visualization_plugin/xy_suggestions.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { partition } from 'lodash';
+import { Position } from '@elastic/charts';
+import { XYArgs } from './xy_expression';
+import { SuggestionRequest, VisualizationSuggestion, TableColumn, TableSuggestion } from '../types';
+
+/**
+ * Generate suggestions for the xy chart.
+ *
+ * @param opts
+ */
+export function getSuggestions(
+  opts: SuggestionRequest<XYArgs>
+): Array<VisualizationSuggestion<XYArgs>> {
+  return Object.entries(opts.tables)
+    .filter(
+      ([_id, { isMultiRow, columns }]) =>
+        // We only render line charts for multi-row queries.
+        isMultiRow &&
+        // We require at least two columns: one for x and at least one for y
+        columns.length > 1 &&
+        // y columns must be numeric
+        columns.find(col => col.operation.dataType === 'number')
+    )
+    .map(([id, table]) => getSuggestionForColumns(id, table));
+}
+
+function getSuggestionForColumns(
+  id: string,
+  table: TableSuggestion
+): VisualizationSuggestion<XYArgs> {
+  const [buckets, values] = partition(
+    prioritizeColumns(table.columns),
+    x => x.operation.isBucketed
```

Comment:
Using "x" as a variable name here is confusing, I suggest using `col` as the variable name